### ISSUE TITLE
Timezone issue

### ISSUE THREAD

**Issue by LukeXF** [winstonjs/winston-daily-rotate-file#88]
Doing a test at 2017-09-08 00:06:15, but the time is outputting an hour behind (UTC?). Problem is moment.js isn't an hour behind (in the same file). 

So when it goes to check for the log file, it sees that the today's log is empty - which is technically right, but both times should line up.

Example messaged placed in yesterday's file:
`[2017-09-07 23:06:15.038] [WARN] message example.`

The transport code:
```
var transport = new winstonMain.transports.DailyRotateFile({
	filename: './log',
	datePattern: 'yyyy-MM-dd.',
	prepend: true,
	level: process.env.NODE_ENV === 'development' ? 'debug' : 'info'
});
```

**Comment by mattberther** [comment]
@LukeXF `winston-daily-rotate-file` does not use the timezone by moment. Depending on the value of the `localTime` option, the timezone used will be either UTC or the timezone of the server running the component.

**Comment by LukeXF** [comment]
I think you misunderstood. I'm aware the timezone used should be the timezone of the server. But I'm saying that value is an hour behind. My server clock was `2017-09-08 00:06:15`.

To back my statement up, I tested some separate code with moment-js and it returned `2017-09-08 00:06:15` whereas your code used an hour earlier. I guess my question is where is UTC/my-server-time selected and what makes that decision?

Apart from that this project is perfect for what I need, great job!

**Comment by mattberther** [comment]
@LukeXF The value of the `localTime` option is evaluated at https://github.com/winstonjs/winston-daily-rotate-file/blob/master/index.js#L745 and then standard javascript code is used to identify the different segments of the date and time. Can you try running similar javascript on your server and let me know what value you receive back?

**Comment by gitramro** [comment]
im having trouble with local time also, im giving localTime value true but im still getting utc on my log file

```
const winston = require('winston');
const winstonRotator = require('winston-daily-rotate-file');

const consoleConfig = [
 new winston.transports.Console({
   'colorize': true
 })
];

const createLogger = new winston.Logger({
 'transports': consoleConfig
});

const successLogger = createLogger;
successLogger.add(winstonRotator, {
 'name': 'access-file',
 'level': 'info',
 'filename': './log/info.log',
 'localTime': true,
 'json': false,
 'datePattern': 'yyyy-MM-dd-',
 'prepend': true
});

module.exports = {
 'successlog': successLogger,
 'errorlog': errorLogger
};

```

Also in my object is displaying the time correctly (hour and minutes)
```

 'access-file':
      EventEmitter {
        domain: null,
        _events: [Object],
        _eventsCount: 1,
        _maxListeners: undefined,
        silent: false,
        raw: false,
        name: 'access-file',
        formatter: undefined,
        level: 'info',
        handleExceptions: false,
        exceptionsLevel: 'error',
        humanReadableUnhandledException: false,
        filename: 'info.log',
        _basename: 'info.log',
        dirname: './log',
        options: [Object],
        json: false,
        colorize: false,
        maxsize: null,
        logstash: null,
        maxFiles: null,
        label: null,
        prettyPrint: false,
        showLevel: true,
        timestamp: true,
        datePattern: 'yyyy-MM-dd-',
        depth: null,
        eol: '\r\n',
        maxRetries: 2,
        prepend: true,
        createTree: false,
        localTime: true,
        zippedArchive: false,
        maxDays: 0,
        _size: 68,
        _created: 0,
        _buffer: [Array],
        _draining: false,
        _failures: 0,
        _archive: false,
        _currentFiles: [Array],
        _year: 2017,
        _month: 9,
        _date: 3,
        _hour: 10,
        _minute: 39,
        _weekday: 'Tue',
        getFormattedDate: [Function],
        _onError: [Function: bound ],
        opening: true },
     'error-file':
```

my log file (at 10:39am on oct 3) 
2017-10-03T17:39:27.879Z - info: Usuario prueba creado por: ramon

Why is my file showing utc time? I checked utc time right now and its the same as the one on my log file

**Comment by mattberther** [comment]
Some long-overdue refactoring of the transport has occurred. A PR is available at #107. Also, a pre-release version has been published to npm. Please `npm install winston-daily-rotate-file@2.0.0-beta` and reopen this issue if the problem remains.

**Comment by milkcoke** [comment]
@LukeXF 
@mattberther 

Even though a long time has passed, here is why daily-rotate-file.js use UTC timestamp 

#### daily-rotate-file.js
```javascript
var loggerDefaults = {

    json: false,
    colorize: false,
    eol: os.EOL,
    logstash: null,
    prettyPrint: false,
    label: null,
    stringify: false,
    depth: null,
    showLevel: true,
   // return UTC time as default
    timestamp: function () {
        return new Date().toISOString();
    }
};
```
